cmd/pilot-agent/main.go

```diff
@@ -119,6 +125,30 @@ var (
 			proxyConfig.StatsdUdpAddress = statsdUDPAddress
 			proxyConfig.ProxyAdminPort = int32(proxyAdminPort)
 
+			var pilotSAN []string
+			switch controlPlaneAuthPolicy {
+			case proxyconfig.AuthenticationPolicy_NONE.String():
+				proxyConfig.ControlPlaneAuthPolicy = proxyconfig.AuthenticationPolicy_NONE
+			case proxyconfig.AuthenticationPolicy_MUTUAL_TLS.String():
+				var ns string
+				proxyConfig.ControlPlaneAuthPolicy = proxyconfig.AuthenticationPolicy_MUTUAL_TLS
+				if serviceregistry == platform.KubernetesRegistry {
```

Comment:
What about consul/eureka? Do we have plan to support them as well?

Reply:
We don't have Istio auth running in those environments.

Reply:
We may still want to secure them - some script generating equivalent certs ( but maybe longer lived since agent is not there), or using the agent may be possible.